gems/aws-sdk-s3/spec/client_spec.rb

```diff
@@ -167,7 +167,8 @@ module S3
           tmpfile.unlink
           s3 = Client.new(stub_responses: true)
           resp = s3.put_object(bucket: 'bucket', key: 'key', body: tmpfile)
-          expect(resp.context.http_request.body_contents).to eq(data)
+          # Need to discuss
+          expect(resp.context.http_request.body.instance_variable_get(:@io).read).to eq(data)
```

Comment:
I realized the body gets replaced with TrailerIO when we use the unsigned body path. I'm wondering if customers ever access this body object directly? If they do, TrailerIO doesn't have all the methods that StringIO  would have, which could break their code.

One option is to swap the body back to the original IO object after we finish sending the request. That way customers would see the original object if they need access later.

Reply:
Fair to worry about breaking someone with this... but it doesn't seem like part of our API contract that we won't change/manipulate the request body during a request.  In fact, It feels like this should be expected and it seems like it would be better for the contents to match what we actually sent rather than a customer provided operation input.